File: 234. Palindrome Linked List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public boolean isPalindrome(ListNode head) {
        ListNode current=head;
        int length=0;
        List<Integer> intlist=new ArrayList<Integer>();
        while(current!=null)
        {
            intlist.add(current.val);
            length++;
            current=current.next;
        }
        int counter=1;
        for(int i=0;i<length/2;i++)
        {
            if(intlist.get(i)!=intlist.get(length-i-1))
            {
                counter=0;
                break;
            }
        }
        if(counter==1)
            return true;
        else
            return false;
    }
}

File: 234. Palindrome Linked List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        current=head
        list1=[]
        length=0
        counter=1
        while(current!=None):
            list1.append(current.val)
            current=current.next
            length=length+1
        for i in range(0,int(length/2)):
            if(list1[i]!=list1[length-1-i]):
                counter=0
                break
        if(counter==1):
            return True
        else:
            return False

